### DISCUSSION TITLE
How to activate the tailhook of the F-15？

### DISCUSSION THREAD

**Discussion by jijnt12345** [JSBSim-Team/jsbsim#1164] [General]
How to activate the tailhook of the F-15 to catch the arrestor wire?

**Comment by seanmcleod** [comment]
You need to provide more details on your setup.

In particular, the first piece of information you need to provide is which F-15 model are you using? For example, the F-15 model included in the JSBSim repo doesn't have a tailhook modelled.

https://github.com/JSBSim-Team/jsbsim/blob/master/aircraft/f15/f15.xml

There could be a number of other F-15 JSBSim models out there. One of the ones I'm familiar with is the one created by Richard Harrison - @Zaretto, which if you search for the word `hook` shows you that a tailhook has been modelled. 

https://github.com/search?q=repo%3AFGMEMBERS%2FF-15%20hook&type=code

The next question would be what environment if any are you running this in? FlightGear, or your own JSBSim based app? Since it doesn't help extending the tail-hook if you don't have an environment that is going to setup and model the arrestor cable.

If you provide more information up-front it means less guessing for people trying to answer 😉

**Reply by jijnt12345** [reply]
I want to use this feature in UE5, not limited to the F-15; any tailhook functionality will do.

**Reply by Zaretto** [reply]
The tricky part of tail hook modelling is the interaction between the arrestor wires and the hook and this isn't part of JSBSim. 

I think this is something that would need to be modelled in the UE interface probably in a similar way to how we do it in FlightGear which is to have the wires as part of the scenery and then populate these into the ground cache. The during the interface to JSBSim the logic will figure out if a wire is engaged based on the hook being on (or below) the ground and intersecting with the wire. 

If the hook is engaged then usually there is an external force element in the model to introduce the appropriate force to the aircraft model at the correct position

**Reply by jijnt12345** [reply]
Where can I find the external force element for the model?

**Comment by seanmcleod** [comment]
@Zaretto exactly, wanted to make @jijnt12345 aware that once you figure out how to lower the tail hook in say your F-15 model, JSBSim doesn't automagically provide any logic for handling the engagement of the hook with some arrestor cable. That has to be provided by the environment you're using the model in.

> not limited to the F-15; any tailhook functionality will do

It's simple for a FDM to provide a tailhook with a mechanism to raise and lower it etc. but 99% of the work is involved in the environment modelling the location of the arrestor, detecting a snag, applying the relevant force etc.

**Reply by jijnt12345** [reply]
Okay, I understand. How can I lower the tailhook in ue5?

**Comment by seanmcleod** [comment]
> Okay, I understand. How can I lower the tailhook in ue5?

You haven't said which aircraft model you're using. Typically a property will be provided that you need to update. But the exact name of the property etc. will depend on the particular aircraft model you're using.

**Reply by jijnt12345** [reply]
f16? I see that the f16 in the JSBSim repository has a tailhook component.

**Comment by seanmcleod** [comment]
So if you've found a tailhook component in the F-16's FDM is it not clear what you need to do in order to lower/raise it?

**Reply by jijnt12345** [reply]
Yes

**Comment by seanmcleod** [comment]
Simply set the property `fcs/hook-engage`.

https://github.com/JSBSim-Team/jsbsim/blob/65618e8d0a4319519e81fdb3500fe00a7291afe5/aircraft/f16/f16.xml#L939-L956

**Reply by jijnt12345** [reply]
ok,thanks

**Reply by jijnt12345** [reply]
Additionally,how to increase the rearward tension (i.e., the tension of the arrestor wire) for the F-16？

**Reply by Zaretto** [reply]
refer to JSBSim\aircraft\f16\Systems\hook.xml

The hook position command is `systems/hook/tailhook-cmd-norm` - set this property to 0..1 where 0 is retracted and 1 is extended

as per the documentation[1][2] for external_reaction  when no `<function>` has been provided in the moment definition, its magnitude
 can be specified through the following property `external_reactions/{moment name}/magnitude-lbsft`. I don't think any of the samples in JSBSim have a fully modelled hook.

For a hook it would be `external_reactions/hook/magnitude-lbsft` that needs to be set to the force to cause deceleration.  It doesn't matter about the hook position - all of that is irrelevant to the application of forces for the external_reaction (f16.xml:231)

You would need to calculate the force yourself; nothing is built into the model or JSBSim to do this.

--------
[1] https://jsbsim-team.github.io/jsbsim/classJSBSim_1_1FGExternalForce.html
[2] https://jsbsim-team.github.io/jsbsim/classJSBSim_1_1FGExternalReactions.html

**Reply by jijnt12345** [reply]
Okay, thank you very much for your guidance

**Comment by seanmcleod** [comment]
In terms of some example code for implementing hook arrestor wire catch detection and applying a retarding force take a look at:

https://github.com/JSBSim-Team/jsbsim/blob/a3101eb6ef7659993bb64c79926fb7bab61be39a/examples/FlightGear.cxx#L1425

**Reply by jijnt12345** [reply]
Okay, thank you very much for your guidance
